File: examples/Poisson/poisson2d/pdeapp.py
# import external modules
import numpy

# import the Exasim frontend (see README, "Using the frontends")
import exasim

# Create pde object and mesh object
pde, mesh = exasim.initializeexasim();

# Define a PDE model: governing equations and boundary conditions
pde['model'] = "ModelD";       # ModelC, ModelD, ModelW
pde['modelfile'] = "pdemodel"; # name of a file defining the PDE model

# Choose computing platform and set number of processors
#pde['platform'] = "gpu";   # choose this option if NVIDIA GPUs are available
pde['mpiprocs'] = 2;        # number of MPI processors
pde['hybrid'] = 1;          # 0 -> LDG, 1 -> HDG

# Set discretization parameters, physical parameters, and solver parameters
pde['porder'] = 3;         # polynomial degree
pde['physicsparam'] = numpy.array([1.0]);   # unit thermal conductivity
pde['tau'] = numpy.array([1.0]);            # DG stabilization parameter
pde['saveSolBouFreq'] = 1;
pde['ibs'] = 1;

# create a mesh of 16 by 16 quads on a square domain
mesh['p'], mesh['t'] = exasim.Mesh.SquareMesh(16,16,1)[0:2];
# expressions for domain boundaries
mesh['boundaryexpr'] = [lambda p: (p[1,:] < 1e-3), lambda p: (p[0,:] > 1-1e-3), lambda p: (p[1,:] > 1-1e-3), lambda p: (p[0,:] < 1e-3)];
mesh['boundarycondition'] = numpy.array([1, 1, 1, 1]); # Set boundary condition for each boundary

# call exasim to generate and run C++ code to solve the PDE model
sol, pde, mesh  = exasim.exasim(pde,mesh)[0:3];

# visualize the numerical solution of the PDE model using Paraview
pde['visscalars'] = ["temperature", 0]; # list of scalar fields for visualization
pde['visvectors'] = ["temperature gradient", numpy.array([1, 2]).astype(int)]; # list of vector fields for visualization
exasim.vis(sol,pde,mesh); # visualize the numerical solution
print("Done!");

File: examples/Advection/GaussianRotating/pdeapp.jl
# import the Exasim frontend (installed via `cmake --install` under
# <prefix>/share/exasim/julia, or Pkg.develop'd from frontends/Julia/Exasim)
using Exasim

# create pde structure and mesh structure
pde, mesh = Exasim.initializeexasim();

# Define PDE model: governing equations, initial solutions, and boundary conditions
pde.model = "ModelC";            # ModelC, ModelD, ModelW
include("pdemodel.jl");          # include the PDE model file

# Set discretization parameters, physical parameters, and solver parameters
pde.porder = 4;          # polynomial degree
pde.torder = 3;          # time-stepping order of accuracy
pde.nstage = 3;          # time-stepping number of stages
pde.physicsparam = [1 1];    # convective velocity
pde.tau = [1.0];               # DG stabilization parameter
pde.dt = 0.025*ones(400);   # time step sizes
pde.soltime = collect(1:length(pde.dt)); # steps at which solution are collected
pde.visdt = 0.025; # visualization timestep size

# Choose computing platform and set number of processors
#pde.platform = "gpu";           # choose this option if NVIDIA GPUs are available
pde.mpiprocs = 4;                # number of MPI processors

# create a linear mesh for a square domain
mesh.p, mesh.t = Exasim.Mesh.SquareMesh(20,20,1); # a mesh of 8 by 8 quadrilaterals
mesh.p = mesh.p .- 0.5; 
# expressions for disjoint boundaries
mesh.boundaryexpr = [p -> (p[2,:] .< -0.5+1e-3), p -> (p[1,:] .> 0.5-1e-3), p -> (p[2,:] .> 0.5-1e-3), p -> (p[1,:] .< -0.5+1e-3)];
mesh.boundarycondition = [1 1 1 1]; # Set boundary condition for each disjoint boundary

# call exasim to generate and run C++ code to solve the PDE model
sol, pde, mesh,~,~,~,~  = Exasim.exasim(pde,mesh);

# visualize the numerical solution of the PDE model using Paraview
pde.visscalars = ["temperature", 1];  # list of scalar fields for visualization
pde.visvectors = []; # list of vector fields for visualization
Exasim.vis(sol,pde,mesh); # visualize the numerical solution
print("Done!");
